### PR TITLE
Docs/135771 Add developer quickstart

### PR DESCRIPTION
**Pull request by AndyRae**
# Changes

Adds a developer quick start to get the app running, with corresponding documentation: https://github.com/HDRUK/CaRROT-Docs/pull/17

Includes:
- Adds a `docker-compose` to get the stack running.
- Updates the `sample-env` and `sample-local-settings` to use default values that work with the docker environment.
- Adds seed data fixtures to populate the `OmopTable` and `OmopField` tables.
- Adds python-dotenv as a dependency, to enable loading configuration from environment variables on all systems.
- Updates readme to point to the docs, it doesn't feel necessary to maintain both.

# Checks

**Important:** please complete these **before** merging.
- [ ] Run migrations, if any.
- [ ] Update `changelog.md`, including migration instructions if any.
- [ ] Run unit tests.

### CODE REVIEW

File: sample-local.settings.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
     "IsEncrypted":false,
     "Values": {
-        "AzureWebJobsStorage":"",
+        "AzureWebJobsStorage":"UseDevelopmentStorage=true",
         "FUNCTIONS_WORKER_RUNTIME":"python",
-        "STORAGE_CONN_STRING":"",
+        "STORAGE_CONN_STRING":"DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;QueueEndpoint=http://azurite:10001/devstoreaccount1;BlobEndpoint=http://azurite:10000/devstoreaccount1;",
```
Comment (dcl10) [inline sample-local.settings.json:6]: no secrets in the repo pls

Comment (AndyRae) [reply sample-local.settings.json:6]: Sorry wasn't clear - this isn't a secret as such, but it is the default connection string for Azurite: https://learn.microsoft.com/en-us/azure/storage/common/storage-use-emulator#authorize-with-shared-key-credentials.
Like I could clear it, and then point to those docs, but I feel a cleaner dev experience is to have it pre-existing - it's no different than having a connection string for a database.

Comment (dcl10) [reply sample-local.settings.json:6]: Oh right is that key the same no matter what then?

Comment (AndyRae) [reply sample-local.settings.json:6]: Yep it's the default to connect to the emulator, so just an default open secret really. See: https://github.com/search?q=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq&type=code

Comment (Jivanam) [reply sample-local.settings.json:6]: Can't we use `UseDevelopmentStorage=true`?

**Pull request merged**